src/ccs/r1cs.rs

```diff
@@ -9,17 +10,65 @@ pub struct R1CS<F: PrimeField> {
     pub B: SparseMatrix<F>,
     pub C: SparseMatrix<F>,
 }
+
 impl<F: PrimeField> R1CS<F> {
     /// returns a tuple containing (w, x) (witness and public inputs respectively)
     pub fn split_z(&self, z: &[F]) -> (Vec<F>, Vec<F>) {
         (z[self.l + 1..].to_vec(), z[1..self.l + 1].to_vec())
     }
+
+    /// check that a R1CS structure is satisfied by a z vector. Only for testing.
+    pub fn check_relation(&self, z: &[F]) -> Result<(), Error> {
+        let Az = mat_vec_mul_sparse(&self.A, z);
+        let Bz = mat_vec_mul_sparse(&self.B, z);
+        let Cz = mat_vec_mul_sparse(&self.C, z);
+        let AzBz = hadamard(&Az, &Bz);
+        assert_eq!(AzBz, Cz);
+
+        Ok(())
+    }
+
+    pub fn relax(self) -> RelaxedR1CS<F> {
```

Comment:
Add docs maybe? Since we make the fn `pub`?

Reply:
Right, agree. Added docs in this amend: https://github.com/privacy-scaling-explorations/folding-schemes/compare/30bb79b36458753b81d2a7b01c1cf47fa7085bfc..fafbe82a35b479ecaba6cc7708ec06e0c102ddaf